code/modules/antagonists/traitor/equipment/Malf_Modules.dm

```diff
@@ -581,6 +581,36 @@ GLOBAL_LIST_INIT(malf_modules, subtypesof(/datum/ai_module))
 		desc = "[initial(desc)] It has [uses] use\s remaining."
 		UpdateButtonIcon()
 
+/// HIGH IMPACT HONKING
+/datum/ai_module/destructive/megahonk
+	name = "Percussive Intercomm Interference"
+	description = "Emit a debilitatingly percussive auditory blast through the station intercoms. Does not overpower hearing protection. Two uses per purchase."
+	cost = 20
+	power_type = /datum/action/innate/ai/honk
+	unlock_text = "<span class='notice'>You upload a sinister sound file into every intercom...</span>"
+	unlock_sound = 'sound/items/airhorn.ogg'
+
+/datum/action/innate/ai/honk
+	name = "Percussive Intercomm Interference"
+	desc = "Rock the station's intercom system with an obnoxious HONK!"
+	button_icon_state = "intercom"
+	uses = 2
+
+/datum/action/innate/ai/honk/Activate()
+	to_chat(owner, span_clown("The intercom system plays your prepared file as commanded."))
+	for(var/obj/item/radio/intercom/found_intercom in GLOB.intercoms_list)
+		if(!found_intercom.on || !found_intercom.listening || found_intercom.wires.is_cut(WIRE_RX)) //Only operating intercoms play the honk
```

Comment:
these vars have spacemandmm decorators forcing you to use getter procs (its required since its the only way to require setter procs). use is_on() and is_listening()